src/geocat/f2py/dpres_plevel_wrapper.py

```diff
@@ -0,0 +1,193 @@
+
+            Examples:
+
+                Example 1: Using dpres_plevel with :class:`xarray.DataArray` input
+
+                .. code-block:: python
+
+                    import numpy as np
+                    import xarray as xr
+                    import geocat.comp
+
+                    # Open a netCDF data file using xarray default engine and load the data stream
+                    ds = xr.open_dataset("./SOME_NETCDF_FILE.nc")
+
+                    # [INPUT] Grid & data info on the source
+                    pressure_surface = ds.PS
+                    pressure_levels = ds.LEV
+                    pressure_top = 0.0
+
+                    # Call the function
+                    result_dp = geocat.comp.dpres_plevel(pressure_levels, pressure_surface, pressure_top)
+            """
+
+    # Apply basic sanity checks on the input data
+    pressure_levels, pressure_surface, pressure_top = sanity_check(pressure_levels,
+                                                                   pressure_surface,
+                                                                   pressure_top
+                                                                   )
+
+    # Inner wrapper call
+    dp = _dpres_plevel(pressure_levels.values,
+                       pressure_surface.values,
+                       pressure_top,
+                       msg_py
+                       )
+
+    # Reshape output based on the dimensionality of pressure_surface
+    if pressure_surface.ndim == 1:
+        dp = dp.reshape(dp.shape[1])
+    elif pressure_surface.ndim == 2:
+        dp = dp.reshape(dp.shape[0] * dp.shape[1], dp.shape[2], dp.shape[3])
+
+    if meta:
+        raise MetaError(
+            "ERROR dpres_plevel: Retention of metadata (other than Xarray.Dataarray.attrs) is not yet supported !")
+
+        # TODO: Retaining possible metadata might be revised in the future
+    else:
+        dp = xr.DataArray(dp, attrs=pressure_surface.attrs)
+
+    return dp
+
+
+# Inner Wrapper _<funcname>()
+# This wrapper basically calls the Fortran function. It also handles
+# transpose of the input and output data as well as missing value
+# representations before and after the Fortran function call.
+
+def _dpres_plevel(plev, psfc, ptop, msg_py):
+
+    # Transpose pressure_surface before Fortran function call
+    if psfc.ndim == 2:
+        psfc = np.transpose(psfc, axes=(1, 0))
+    elif psfc.ndim == 3:
+        psfc = np.transpose(psfc, axes=(2, 1, 0))
+
+    # Handle Python2Fortran missing value conversion
+    psfc, msg_py, msg_fort = py2fort_msg(psfc, msg_py=msg_py)
+
+    # Fortran function call
+    dp = dpresplvl(plev, psfc, np.float64(1000), pmsg=msg_fort)
+
+    # Transpose output to corect dimension order before returning it to outer wrapper
+    dp = np.asarray(dp)
+    dp = np.transpose(dp, axes=(3,2,1,0))
+
+    # Handle Fortran2Python missing value conversion back
+    fort2py_msg(dp, msg_fort=msg_fort, msg_py=msg_py)
+
```

Comment:
Need to also revert missing value swapping from psfc on line 131. Otherwise we would have silently modified the contents of that input array. This could lead to a very confused user.

Reply:
Sure, thanks for great catch!

Reply:
Marking as resolved per commit 5cb4796